### DISCUSSION TITLE
Kimai Plugins and AI LLM

### DISCUSSION THREAD

**Discussion by rmab-se** [kimai/kimai#4458] [Plugins]
As a non coder but understanding basics of coding I am currently exploring LLM AI and possibility of contributing with it.

has anyone had any progress in making a functional plugin with LLM ?

I am currently using Google bard to dig in Kimai’s API with the goal of making a plugin that could make it easy to change templates for export and Invoice.

**Comment by kevinpapst** [comment]
Super interesting topic. But I have no idea what you want to know?
Any working plugin is a foundation to an LLM API. Just call it with your prompt and an HTTP request.
What is your concrete question?

**Comment by rmab-se** [comment]
Is there a known way to use AI chat to generate plugins. And to implement them into kimai. I’m exploring this subject but without sucess.. 

And I fully understand that the codes provided by ai is not always great. But if one could generate a base code and debug it from there.

**Reply by kevinpapst** [reply]
There is a demo bundle that you can use as starter: https://github.com/kimai/DemoBundle
